### ISSUE TITLE
Non-serializable attribute org.atmosphere.cpr.BroadcasterFactory

### ISSUE THREAD

**Issue by bpapez** [Atmosphere/atmosphere#2129]
We are currently testing whether our web application can be used with externalizing Http session data to a database (like Redis). We are getting this exception:

```
java.lang.IllegalArgumentException: setAttribute: Non-serializable attribute org.atmosphere.cpr.BroadcasterFactory
    at org.apache.catalina.session.StandardSession.setAttribute(StandardSession.java:1480)
    at org.apache.catalina.session.StandardSession.setAttribute(StandardSession.java:1441)
    at com.orangefunction.tomcat.redissessions.RedisSession.setAttribute(RedisSession.java:59)
    at org.apache.catalina.session.StandardSessionFacade.setAttribute(StandardSessionFacade.java:142)
    at org.atmosphere.cpr.AsynchronousProcessor.action(AsynchronousProcessor.java:152)
    at org.atmosphere.cpr.AsynchronousProcessor.suspended(AsynchronousProcessor.java:108)
    at org.atmosphere.container.Servlet30CometSupport.service(Servlet30CometSupport.java:66)
    at org.atmosphere.cpr.AtmosphereFramework.doCometSupport(AtmosphereFramework.java:2257)
    at org.atmosphere.cpr.AtmosphereServlet.doPost(AtmosphereServlet.java:190)
    at org.atmosphere.cpr.AtmosphereServlet.doGet(AtmosphereServlet.java:176)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:622)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
```

Currently I am not able to see what is the consequence or side effect to this exception or how to solve it. Would it be possible to make the DefaultBroadcasterFactory serializable or if not, to not store it in session? I see that it is only used in SessionSupport.sessionDestroyed to set the session in atmosphere resources to null. Any alternative recommendation how to solve that would be very welcome.

**Comment by bpapez** [comment]
The issue is a side effect of: https://github.com/Atmosphere/atmosphere/issues/1762
I have added two comments in the commit: https://github.com/Atmosphere/atmosphere/commit/e6cc132f509ca2b3a675697b4db3f824bd916d5b#diff-5b6554e9c01f6234c317bf3588c30bfdR46

**Comment by oyvindhorneland** [comment]
@bpapez If https://github.com/Jahia/atmosphere/commit/50151f6ed955a418c1557b5d7b8cd879eec53ce7 worked as expected, would it be possible for you to create a pull request to Atmosphere for it? I too bumped into this issue when combining Atmosphere with Spring Session.

**Comment by bpapez** [comment]
@oyvindhorneland: Actually we were running into multiple issues with the combination of Spring Session and Atmosphere (see also #2133 ). 

There is for instance also the limitation, that we no longer can activate the usage of websockets. The reason is that Atmosphere with websockets creates HTTP sessions too early in the request processing (in the requestInitialized event), so the Spring Session filters were not called yet and thus regular Tomcat sessions are created and remove the session cookies created by Spring Session. 

However SSE and long polling were tested and work fine with some custom workaround code in Atmosphere.

**Comment by oyvindhorneland** [comment]
@bpapez Thank you for the update. 

Yes, I see that the ContainerInitializer adds a ServletRequestListener that extracts the session, so a fix for that approach would then need be able to call Spring Session somehow (without filter support)..

**Comment by jfarcand** [comment]
@bpapez If you have the cycle, just do me a favor and send a pull request with your changes.

**Comment by glucas** [comment]
Any update on this issue?

**Comment by jfarcand** [comment]
Fixed #2337
